src/main/java/com/example/springbootbookshop/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.example.springbootbookshop.service.impl;
+
+import com.example.springbootbookshop.dto.category.CategoryDto;
+import com.example.springbootbookshop.dto.category.CategoryRequestDto;
+import com.example.springbootbookshop.entity.Category;
+import com.example.springbootbookshop.exception.EntityNotFoundException;
+import com.example.springbootbookshop.mapper.CategoryMapper;
+import com.example.springbootbookshop.repository.CategoryRepository;
+import com.example.springbootbookshop.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll() {
+        return categoryRepository.findAll().stream()
+                .map(category -> categoryMapper.toDto(category))
```

Comment:
you can use method reference here
```suggestion
                .map(categoryMapper::toDto)
```